enterprise/server/scheduling/priority_queue/priority_queue_test.go

```diff
@@ -0,0 +1,75 @@
+package priority_queue_test
+
+import (
+	"fmt"
+	"math/rand/v2"
+	"slices"
+	"testing"
+
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/scheduling/priority_queue"
+	"github.com/stretchr/testify/require"
+
+	scpb "github.com/buildbuddy-io/buildbuddy/proto/scheduler"
+)
+
+func TestIndexedOperations(t *testing.T) {
```

Comment:
nit: I think adding a simple test would also be useful here -- where you just push some stuff in and pop it back out and ensure it's in the correct order.